jetty-core/jetty-osgi/src/main/java/org/eclipse/jetty/osgi/OSGiDeployableBundleMetadata.java

```diff
@@ -115,139 +130,90 @@ private static String getContextPath(Bundle bundle)
 
         return contextPath;
     }
-    
+
     /**
-     * @param manager the DeploymentManager to which to deploy
-     * @param provider the provider that discovered the context/webapp
-     * @param bundle the bundle associated with the context/webapp
+     * Register the Jetty deployed context/webapp as a service, as
+     * according to the OSGi Web Application Specification.
      */
-    public OSGiApp(DeploymentManager manager, AppProvider provider, Bundle bundle)
-    throws Exception
-    {
-        super(manager, provider, getBundlePath(bundle));
-
-        _bundle = Objects.requireNonNull(bundle);
-        _bundleResource = getBundleAsResource(bundle);
-        
-        //copy selected bundle headers into the properties
-        Dictionary<String, String> headers = bundle.getHeaders();
-        Enumeration<String> keys = headers.keys();
-        while (keys.hasMoreElements())
+    public static void registerAsOSGiService(ContextHandler contextHandler)
```

Comment:
This doesn't feel like the right place for this method as this class is purely about the metadata associated with the `Bundle` that will become a `ContextHandler`.  Needs to be moved to some `Util` class.

Reply:
Ditto to above, is `OSGiContextHandlerUtil` ok?

Reply:
There is already a `Util` class  in `org.eclipse.jetty.osgi.util.Util` - can you add it to that?

Reply:
Done